front/app/components/ProjectCard/index.tsx

```diff
@@ -374,11 +374,14 @@ const ProjectCard = memo<InputProps>(
 
     // We use this hook instead of useProjectImages,
     // because that one doesn't work with our caching system.
-    const { data: projectImage } = useProjectImage({
+    const imageId = project?.data.relationships.project_images?.data[0]?.id;
+    const { data: _projectImage } = useProjectImage({
       projectId,
-      imageId: project?.data.relationships.project_images?.data[0]?.id,
+      imageId,
     });
 
+    const projectImage = imageId ? _projectImage : undefined;
```

Comment:
Without this, if the project changes, `imageId` might suddenly be `undefined` because there is no image. But `react-query` then just keeps the previous image belonging to a different project.